### DISCUSSION TITLE
[foundry-zksync] Unable to run tests

### DISCUSSION THREAD

**Discussion by Carolina-Pinheiro** [zkSync-Community-Hub/zksync-developers#85] [General] [answered]
### Environment

Testnet

### Provide a brief description of the functionality you're trying to implement.

I'm trying to run a test using zkforge with the command 'zkforge test', however I get an error when running any test

> The application panicked (crashed).
> Message:  Unable to read file: Os { code: 2, kind: NotFound, message: "No such file or directory" }
> Location: cli/src/cmd/forge/test/mod.rs:501
> 
> This is a bug. Consider reporting it at https://github.com/foundry-rs/foundry
> 
> Backtrace omitted. Run with RUST_BACKTRACE=1 environment variable to display it.
> Run with RUST_BACKTRACE=full to include source snippets.

### What is the specific issue or error you're encountering?

By digging through the rust code, It seems that a single contract is hard-coded in the test.rs file.

![image](https://github.com/zkSync-Community-Hub/zkync-developers/assets/57190987/0875729f-297f-40d8-9f74-7600f299a335)

### Can you share the error messages or logs you're receiving, if any?

> The application panicked (crashed).
> Message:  Unable to read file: Os { code: 2, kind: NotFound, message: "No such file or directory" }
> Location: cli/src/cmd/forge/test/mod.rs:501
> 
> This is a bug. Consider reporting it at https://github.com/foundry-rs/foundry
> 
> Backtrace omitted. Run with RUST_BACKTRACE=1 environment variable to display it.
> Run with RUST_BACKTRACE=full to include source snippets.

### Have you made any recent changes to the contract before encountering this issue?

No.

### Are there any external libraries or contracts that your contract interacts with?

I'm using foundry-zksync: https://github.com/matter-labs/foundry-zksync

### Can you provide the relevant portions of your contract code where the issue is occurring?

It's with any test contract using foundry.

### Have you tried to isolate the problem, and if so, what were the results?

The problem seems to be that the test has a specific contract hard-coded.

### What steps have you already taken to try to resolve the issue?

Looking for similar issues in the foundry-zksync repo (there is an user with the same [problem](https://github.com/matter-labs/foundry-zksync/issues/105), but with no answer)
Asking for help in the zkSync discord
Looking for the same error in the foundry support telegram group

**Comment by dutterbutter** [comment, marked as answer]
Hey @Carolina-Pinheiro 👋 

Thanks for checking out zkfoundry! It is still currently in alpha version and running `zkforge test` is currently not supported. We are working to iteratively add support for all the features you expect from foundry. We will update the README so its far more clearer what is supported right now, what is actively being worked on, and what is not supported. Thank you very much for opening the post 💯

**Comment by vatosherrod** [comment]
It seems that zkforge is crashing due to a hardcoded contract path in the test.rs file. You may need to modify the test file to point to the correct contract path or provide the necessary contract file. Additionally, consider reporting this issue to the foundry project maintainers for further assistance and resolution.

**Comment by beckyhaynes** [comment]
It seems that the issue you're encountering is related to a bug in the zkforge testing framework, where a specific contract file is hard-coded in the test module. This results in the test failing to read the file and crashing the application.

To resolve this issue, you can try the following steps:

Check for Updates: Ensure that you are using the latest version of zkforge and any related dependencies. There might be a newer version that includes a fix for this bug.

Temporary Workaround: As a temporary workaround, you can try modifying the test module to specify the correct contract file that you want to test. This might involve modifying the test.rs file or any relevant configuration files.

Report the Issue: Since this seems to be a bug in the zkforge framework, consider reporting the issue to the maintainers on their GitHub repository. Provide detailed information about the error and steps to reproduce it so that they can investigate and fix it.

Alternative Testing Framework: If the issue persists and you need to run tests urgently, consider using an alternative testing framework or tool until the zkforge issue is resolved.

Community Support: Continue seeking help from the zkSync Discord channel, the Foundry support Telegram group, or any other relevant community forums. Sometimes, other users might have encountered the same issue and found a workaround or solution.

By following these steps and actively engaging with the community and maintainers, you should be able to resolve the issue with zkforge testing.

**Comment by hanjv** [comment]
The issue seems to be related to a specific contract hard-coded in the test.rs file of zkforge. It's recommended to report this bug to the maintainers of zkforge or foundry-rs/foundry repository on GitHub for further assistance and resolution.

**Comment by jwitte3** [comment]
It seems like there is a bug in zkforge's test module where a specific contract file is hard-coded, causing the application to panic when running tests. Consider reporting this issue to the project maintainers on GitHub to get assistance and resolution.

**Comment by davehutchinsv** [comment]
The user is encountering an error message stating "Unable to read file: Os { code: 2, kind: NotFound, message: "No such file or directory" }" when using foundry-zksync in any test contract. They have not made any recent changes to the contract and have tried to isolate the problem by looking for similar issues in the foundry-zksync repo and asking for help in the zkSync discord and foundry support telegram group. Further investigation and debugging may be needed to determine the cause of the issue and the feasibility and impact of any necessary changes
